includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -1174,10 +1174,29 @@ public function add_last_activity_to_user_query( WP_User_Query $query ) {
 			WHERE {$wpdb->comments}.user_id = {$wpdb->users}.ID
 			AND {$wpdb->comments}.comment_approved IN ('complete', 'passed', 'graded')
 			AND {$wpdb->comments}.comment_type = 'sensei_lesson_status'
-			ORDER BY {$wpdb->comments}.comment_date_gmt DESC
 		) AS last_activity_date";
 	}
 
+	/**
+	 * Order query based on the custom field.
+	 *
+	 * @since  4.3.0
+	 * @access public
+	 *
+	 * @param WP_User_Query $query The user query.
+	 */
+	public function add_orderby_custom_field_to_query( WP_User_Query $query ) {
+		global $wpdb;
+		if ( ! isset( $query->query_vars['orderby'] ) || ! isset( $query->query_vars['order'] ) ) {
+			return '';
+		}
```

Comment:
I'm not sure if this check is needed because you can safely assume that the `orderby` and `order` query vars are always set.

Reply:
Well, I've added it to be sure, since the error is tromenduous when it happens and I wasn't 100% sure there isn't any case when this happens.

Reply:
The problem with defensive programming is that you (or other devs) won't notice if things are breaking.

For example, in your case, the `orderby` is always set by `WP_User_Query` and if another developer (or a third-party plugin) unsets the `orderby` for some reason, this code should throw an error, because this is unexpected and it breaks the business logic. When there is an error, it will be easier to notice, debug and fix. If there is no error, a user will report that the sorting is not working and we won't have an idea (or have to spend hours debugging) why it's not working in their case.

My point is that in the long term, defensive programming can introduce more bugs than it prevents and those bugs require more time to fix.

Reply:
Yeah, makes sense when you put it like that. Sorry I made a change in the same PR as the other change, don't want to force push it now if I make 2 commits. But here it is. [here](update/fix_sorting_for_students_column)